src/main/java/org/apache/geronimo/microprofile/impl/jwtauth/jwt/KidMapper.java

```diff
@@ -120,7 +181,44 @@ private String tryLoad(final String value) {
             throw new IllegalArgumentException(e);
         }
 
-        // else direct value
+        // load jwks via url
+        if (jwksUrl != null) {
+            if(reloadJwksRequest != null) {
+                try {
+                    reloadJwksRequest.get();
+                } catch (InterruptedException | ExecutionException e) {
```

Comment:
split `InterruptedException` and `Thread.currentThread().interrupt()` there ?

catch of `ExecutionException` should `throw e.getCause() instanceof RuntimeException ? (RuntimeException) e.getCause() : new IllegalStateException(e.getCause());` probably